distribution/pull_v2.go

```diff
@@ -356,6 +360,12 @@ func (p *v2Puller) pullV2Tag(ctx context.Context, ref reference.Named) (tagUpdat
 		return false, fmt.Errorf("image manifest does not exist for tag or digest %q", tagOrDigest)
 	}
 
+	if m, ok := manifest.(*schema2.DeserializedManifest); ok {
+		if m.Manifest.Config.MediaType == schema2.MediaTypePluginConfig {
```

Comment:
That we know about. At some point we will have to be restrictive in the supported values. If it fails then we can assess whether it is legitimate and we should make a code change or whether the manifest needs to be recreated. We accept the `application/octet-stream` because we know there were many manifests mistakenly created with it and that at the time when they were created the only supported value is `application/vnd.docker.container.image.v1+json`.

For this change in particular, since now there are multiple possible types, I think it makes sense to define what is supported and have a special case for that one distribution bug.